### PR TITLE
Add support for the Community Slackbot

### PR DESCRIPTION
**Pull request by johansolbakken**
This PR introduces the `slack_message` schema. It features `SlackMessageSearcher` and `ThreadSearcher` which uses group by to retrieve the conversations with highest relevance. 

Four new query profiles have been added:
- `slackmessage` that runs `SlackMessageSearcher`.
- `thread` that runs `ThreadSearcher`.
- `combinedllmsearch` that combines result from `thread` and `llmsearch` query profiles.
- `combinedragsearch` that uses `RAGSearch` with the combined results from `thread` and `llmsearch`.

### CODE REVIEW

File: src/main/application/schemas/slack_message.sd
Comment (jobergum) [inline src/main/application/schemas/slack_message.sd:10]: Add match:word to avoid tokenization

File: src/main/application/schemas/slack_message.sd
Comment (jobergum) [inline src/main/application/schemas/slack_message.sd:14]: Add match:word to avoid tokenization, see [match](https://docs.vespa.ai/en/reference/schema-reference.html#match)

File: src/main/application/schemas/slack_message.sd
Comment (jobergum) [inline src/main/application/schemas/slack_message.sd:15]: Remove this

File: src/main/application/schemas/slack_message.sd
Comment (jobergum) [inline src/main/application/schemas/slack_message.sd:19]: Remove attribute unless you want to group or sort on text.

**Review by jobergum — CHANGES_REQUESTED** [review]
Great! Please see the requested changes in the schema. Code is easy to read.

File: src/main/application/schemas/slack_message.sd
Comment (jobergum) [reply src/main/application/schemas/slack_message.sd:15]: I mean remove stemming:best

File: src/main/application/schemas/slack_message.sd
Comment (johansolbakken) [reply src/main/application/schemas/slack_message.sd:15]: nice, done

File: src/main/application/schemas/slack_message.sd
Comment (johansolbakken) [reply src/main/application/schemas/slack_message.sd]: ok thanks, i removed it

**Review by jobergum — APPROVED** [review]

**Comment by johansolbakken** [comment]
@jobergum  I don't have write access, so can't merge this

**Pull request merged**
